Added Player Avatar Overlay to Lobby

ODE-BOY opened the pull request:
The image of the players roll will be overlayed onto their position on the page.
If a detective respects a mafia the mafia image will be overlayed onto the suspected player.

# Description
This Draft PR is Fixing #317 

This PR will overlay a characters hat avatar over the player's respective avatar.
It will also overlay the mafia character over the detectives suspect if said suspect is a mafia.

# Demo

Before: [Insert Screenshot]
![image](https://user-images.githubusercontent.com/48495086/114791654-f2fc8980-9dda-11eb-9efa-e3fd538d1dbd.png)

After: [Insert Screenshot]
![image](https://user-images.githubusercontent.com/48495086/114982215-46f29580-9ee3-11eb-90ee-a4914dc9d05b.png)
![image](https://user-images.githubusercontent.com/48495086/114982309-612c7380-9ee3-11eb-9dbb-198c2c4e46c5.png)
![image](https://user-images.githubusercontent.com/48495086/114982549-b5cfee80-9ee3-11eb-944c-df22d78a9423.png)
![image](https://user-images.githubusercontent.com/48495086/114982655-d730da80-9ee3-11eb-88ea-3d3d36e5f343.png)



# Type of change:
* New feature (Non-breaking change that adds functionality)

# Checklist:
* Have you merged main into your branch?
* Have you tested your changes to ensure it works as expected and does not break existing functionality?
* If applicable, please ensure sufficient tests are added that is related to the changes.
* Please ensure this PR has a `label`, is linked to an `issue` and is related to a `project`
* If new documentation is required for this change, have you created a new documentation issue that describes the documentation needed?

ODE-BOY committed:
Added Player Avatar Overlay to Lobby

The image of the players roll will be overlayed onto their position on the page.
If a detective respects a mafia the mafia image will be overlayed onto the suspected player.

JChamberlain28 commented:
based on the approach taken for #305 you might want to only overlay the hats onto the existing avatars :)

JChamberlain28 reviewed (changes requested):
Please see the closed PR #336 . That does not use overlaying for avatars. This impacts this draft PR. A decision must be made on if this work is to be dropped or continued due to the time constraint. Excellent work so far though.

ODE-BOY committed:
Replace the avatar overlay with a character hat overlay

Replaced the avatar character overlay with a character hat overlay that will be imposed on the head of each avatar.

ODE-BOY committed:
Merge branch 'a2-main' into feature/role-avatars

JChamberlain28 commented on frontend/src/Components/Player.jsx:
```diff
@@ -62,6 +67,15 @@ export default function Player({ playerName, style, childRef }) {
+        [mafiaImage]: (gameState.role === 'mafia'),
+        [medicImage]: (gameState.role === 'medic'),
+    })
+
```
This is a really clean way of doing it. nice!

JChamberlain28 reviewed (approved):
Tested and it works on my pc with 1080p and normal display scaling. Might be worth someone testing on a high DPI screen e.g. a mac or high end windows laptop.

The approach taken is reasonable.
I appreciate the hard work and fast action, bearing in mind this issue was blocked for a bit. 

Good work!

AngelaLH reviewed (approved):
Looks awesome and matches the design :D There are some instances where it may be appropriate to have the hats showing e.g. on the end screen when the jester won they didn't have their hat, but we can make that a future issue as it wasn't stated in this one.  It's working on my 4K screen but haven't tested on a larger monitor. Thanks for your hard work!
<img width="256" alt="chrome_OKpht9Ynij" src="https://user-images.githubusercontent.com/41172492/115089465-6088f100-9f66-11eb-97b1-92132876d724.png">

AngelaLH reviewed (commented):
Just a quick thing - on the role reveal screen only the citizen has a smiley face. Was this the case before your changes? Might want to remove your changes to Medic.svg etc as they are not used in your code, otherwise, all is good :)

JChamberlain28 commented:
> Just a quick thing - on the role reveal screen only the citizen has a smiley face. Was this the case before your changes? Might want to remove your changes to Medic.svg etc as they are not used in your code, otherwise, all is good :)

This applies to issue #315, the PR for that issue shows a jester with a smiley face so I think this change comes in this PR, #335.

ODE-BOY committed:
Merge branch 'a2-main' into feature/role-avatars

ODE-BOY commented:
> 
> 
> > Just a quick thing - on the role reveal screen only the citizen has a smiley face. Was this the case before your changes? Might want to remove your changes to Medic.svg etc as they are not used in your code, otherwise, all is good :)
> 
> This applies to issue #315, the PR for that issue shows a jester with a smiley face so I think this change comes in this PR, #335.



> 
> 
> > Just a quick thing - on the role reveal screen only the citizen has a smiley face. Was this the case before your changes? Might want to remove your changes to Medic.svg etc as they are not used in your code, otherwise, all is good :)
> 
> This applies to issue #315, the PR for that issue shows a jester with a smiley face so I think this change comes in this PR, #335.

Does this mean this is safe to merge?
